tests/sentry/api/helpers/test_default_symbol_sources.py

```diff
@@ -0,0 +1,110 @@
+from sentry.api.helpers.default_symbol_sources import set_default_symbol_sources
+from sentry.testutils.cases import TestCase
+
+
+class SetDefaultSymbolSourcesTest(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.organization = self.create_organization(owner=self.user)
+
+    def test_no_platform(self):
+        """Projects without a platform should keep their epoch defaults"""
+        project = self.create_project(organization=self.organization, platform=None)
+        # Capture epoch defaults before calling set_default_symbol_sources
+        epoch_defaults = project.get_option("sentry:builtin_symbol_sources")
+
+        set_default_symbol_sources(project, self.organization)
+
+        # Should not change the defaults for projects without a platform
+        sources = project.get_option("sentry:builtin_symbol_sources")
+        assert sources == epoch_defaults
+
+    def test_unknown_platform(self):
+        """Projects with unknown platforms should keep their epoch defaults"""
+        project = self.create_project(organization=self.organization, platform="unknown-platform")
+        # Capture epoch defaults before calling set_default_symbol_sources
+        epoch_defaults = project.get_option("sentry:builtin_symbol_sources")
+
+        set_default_symbol_sources(project, self.organization)
+
+        # Should not change the defaults for projects with unknown platforms
+        sources = project.get_option("sentry:builtin_symbol_sources")
+        assert sources == epoch_defaults
```

Comment:
should we test if the epoch changes, if a known platform is provided?

Reply:
This is already covered by the existing tests:
test_electron_platform (line 51) - verifies electron gets ios, microsoft, electron
test_unity_platform (line 62) - verifies unity gets ios, microsoft, android, nuget, unity, nvidia, ubuntu
These tests implicitly verify epoch defaults are overridden because they assert platform-specific sources are present.